apps/sms/views/shared/js/utils.js

```diff
@@ -711,6 +711,60 @@
     },
 
     /**
+     * Returns a rate limited function, useful to reduce the number of function
+     * calls when listening to user input (keypress) or browser events.
+     * @params {function} func - The function that should be rate-limited
+     * @params {number} delay - Min amount of time (ms) between two calls
+     * @param {Boolean} preventfirstCall - Prevent first synchronous call when
+     * true.
+     * @param {Boolean} preventLastCall - Prevent last trailing call when true.
+     * @returns {function} - The rate limited function that will call the
+     *  original function
+     */
+    throttle: function(func, delay, preventfirstCall, preventLastCall) {
```

Comment:
Seems like you can use [default parameter](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/default_parameters) if your delay / preventfirstCall / preventLastCall rarely apply the value other than default 300ms/ true /false